.github/workflows/nvidia-test.yml

```diff
@@ -0,0 +1,70 @@
+name: Nvidia Graphics Tests
+
+on:
+    workflow_run:
+      workflows:
+        - "Smoke Test"
+      types:
+        - completed
+    push: # OBS; To be Removed latter, used only to check workflow functionallity
+      branches: [ main, nvidia-ci-workflow ] # TODO; Remove this latter
+    pull_request: # TODO; Remove this latter
+      branches: [ main ] # TODO; Remove this latter
+
+jobs:
+
+    build-and-upload-artifact:
+        runs-on: [ubuntu-latest]
+        env:
+            PR_NUMBER: ${{ github.event.number }}
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+            
+            - name: Setup upterm session
+              uses: lhotari/action-upterm@v1
+              with:
+                wait-timeout-minutes: 2
+
+            - name: Build docker snap
+              uses: snapcore/action-build@v1
+              id: snapcraft
+              with:
+                snapcraft-args: snap --output docker.snap
+            
+            - name: Install snapcraft
+              run: |
+                sudo snap install snapcraft --classic
+            
+            - name: Upload artifact to snapstore branch
+              run: |
+                export SNAPCRAFT_STORE_CREDENTIALS="${{ secrets.SNAPCRAFT_LOGIN }}"
+                snapcraft upload ${{steps.snapcraft.outputs.snap}} --release latest/edge/pr-${{ env.PR_NUMBER }}
```

Comment:
@lucaskanashiro the plan is to have this secret to login in the CI machine and upload to a temporary branch.